File: server/makc2022--nodejs--nestjs/src/layer3/nosql-mongo/sample/entities/dummy-one-to-many/dummy-one-to-many-entity.loader.ts
/** Copyright (c) 2022 Maxim Kuzmin. All rights reserved. Licensed under the MIT License. */

import { nameof } from '../../../../../layer1/common/common-functions';
import { EntityLoader } from '../../../../../layer1/entity/entity-loader';
import { DummyOneToManyEntityObject } from './dummy-one-to-many-entity.object';

/** Загрузчик сущности "DummyOneToMany". */
export class DummyOneToManyEntityLoader extends EntityLoader<DummyOneToManyEntityObject> {
  /** @inheritdoc */
  constructor(entityObject?: DummyOneToManyEntityObject) {
    super(entityObject ?? ({} as DummyOneToManyEntityObject));
  }

  /** @inheritdoc */
  override load(
    entityObject: DummyOneToManyEntityObject,
    loadableProperties?: Set<string>
  ): Set<string> {
    const result = super.load(entityObject, loadableProperties);

    if (result.has(nameof<DummyOneToManyEntityObject>('id'))) {
      this.entityObject.id = entityObject.id;
    }

    if (result.has(nameof<DummyOneToManyEntityObject>('name'))) {
      this.entityObject.name = entityObject.name;
    }

    return result;
  }

  /** @inheritdoc */
  protected override createAllPropertiesToLoad(): Set<string> {
    return new Set<string>([
      nameof<DummyOneToManyEntityObject>('id'),
      nameof<DummyOneToManyEntityObject>('name'),
    ]);
  }
}

File: server/makc2022--dotnet/Makc2022.Layer3.Sql.Sample/Entities/DummyOneToMany/DummyOneToManyEntityLoader.cs
// Copyright (c) 2022 Maxim Kuzmin. All rights reserved. Licensed under the MIT License.

using Makc2022.Layer3.Sql.Sample.Entity;

namespace Makc2022.Layer3.Sql.Sample.Entities.DummyOneToMany
{
    /// <summary>
    /// Загрузчик сущности "DummyOneToMany".
    /// </summary>
    public class DummyOneToManyEntityLoader : EntityLoader<DummyOneToManyEntityObject>
    {
        #region Constructors

        /// <inheritdoc/>
        public DummyOneToManyEntityLoader(DummyOneToManyEntityObject? entityObject = null)
            : base(entityObject ?? new DummyOneToManyEntityObject())
        {
        }

        #endregion Constructors

        #region Public methods

        /// <inheritdoc/>
        public sealed override HashSet<string> Load(DummyOneToManyEntityObject entityObject, HashSet<string>? loadableProperties = null)
        {
            var result = base.Load(entityObject, loadableProperties);

            if (result.Contains(nameof(EntityObject.Id)))
            {
                EntityObject.Id = entityObject.Id;
            }

            if (result.Contains(nameof(EntityObject.Name)))
            {
                EntityObject.Name = entityObject.Name;
            }

            return result;
        }

        #endregion Public methods

        #region Protected methods

        /// <inheritdoc/>
        protected override HashSet<string> CreateAllPropertiesToLoad()
        {
            return new HashSet<string>
            {
                nameof(EntityObject.Id),
                nameof(EntityObject.Name)
            };
        }

        #endregion Protected methods
    }
}
